pkg/cli/image/mirror/mirror.go

```diff
@@ -178,8 +184,23 @@ func NewCmdMirrorImage(streams genericclioptions.IOStreams) *cobra.Command {
 	return cmd
 }
 
+// This needs to run before FilterOptions.Complete because filter is set to system os/arch if/when --filter-by-os flag is not set
+// and also before MirrorImageOptions.Complete since --filter-by-os is set to wildcard if/when --keep-manifest-list=true
+func (o *MirrorImageOptions) Validate() error {
+	if o.KeepManifestList && len(o.FilterOptions.FilterByOS) > 0 && !o.FilterOptions.IsWildcardFilter() {
+		return fmt.Errorf("cannot preserve the manifest list digest while also filtering its constituents. --keep-manifest-list=true cannot be passed with --filter-by-os, unless --filter-by-os=.* \nAlso, --keep-manifest-list=true is equivalent to --filter-by-os=.*, so no need to pass both flags.")
+	}
+	return nil
+}
+
 func (o *MirrorImageOptions) Complete(cmd *cobra.Command, args []string) error {
-	if err := o.FilterOptions.Complete(cmd.Flags()); err != nil {
+	if o.KeepManifestList {
+		o.FilterOptions.FilterByOS = ".*"
+	}
+
+	o.FilterOptions.Complete(cmd.Flags())
+
+	if err := o.FilterOptions.Validate(); err != nil {
 		return err
```

Comment:
This has to run _after_ complete, otherwise the FilterOptions.OSFilter is not set correctly (due to the defaulting that happens in FilterOptions.Complete)